lib/schema-processors.js

```diff
@@ -0,0 +1,268 @@
+'use strict';
+
+const { Enum, iter } = require('@metarhia/common');
+
+const {
+  getCategoryType,
+  getReferenceType,
+  extractDecorator,
+  extractByPath,
+} = require('./schema-utils');
+
+const { SchemaValidationError, MetaschemaError } = require('./schema-errors');
+
+// Verifies that there could be link from source category to destination
+//   source - <Object>
+//   sourceName - <string>
+//   destination - <Object>
+//   destinationName - <string>
+//   propertyName - <string>
+// Returns: <SchemaValidationError> | <null> information about error or null
+//          if link is valid
```

Comment:
Incorrect indentation:
```suggestion
//     if link is valid
```